src/history.js

```diff
@@ -369,7 +371,7 @@ class History extends Emitter {
    * Set the limit of the history object.
    */
   setLimit(limit: number) {
-    this.limit = Math.max(0, limit)
+    this.limit = Math.max(DEFAULTS.maxHistory, limit)
```

Comment:
This is a fun one. We always want the minimum limit to be 1 right now, so that there's always a "parent" action to derive a base state snapshot from.